tests/commands/test_build.py

```diff
@@ -0,0 +1,692 @@
+    # change dir to the temp one, where we will have the 'dir1/dir2' tree
+    dir1 = tmp_path / 'dir1'
+    dir1.mkdir()
+    dir2 = dir1 / 'dir2'
+    dir2.mkdir()
+    monkeypatch.chdir(tmp_path)
+
+    dir2 = pathlib.Path(str(dir2))  # comparisons below don't work well in Py3.5
+    validator = Validator()
+    resp = validator.validate_from('dir1/dir2')
+    assert resp == dir2
+
+
+def test_validator_from_expanded():
+    """'from' param: expands the user-home prefix."""
+    validator = Validator()
+    resp = validator.validate_from('~')
+    assert resp == pathlib.Path('~').expanduser().absolute()
+
+
+def test_validator_from_exist():
+    """'from' param: checks that the directory exists."""
+    validator = Validator()
+    expected_msg = "the charm directory was not found: '/not_really_there'"
+    with pytest.raises(CommandError, match=expected_msg):
+        validator.validate_from('/not_really_there')
+
+
+def test_validator_from_isdir(tmp_path):
+    """'from' param: checks that the directory is really that."""
+    testfile = tmp_path / 'testfile'
+    testfile.touch()
+
+    validator = Validator()
+    expected_msg = "the charm directory is not really a directory: '{}'".format(testfile)
+    with pytest.raises(CommandError, match=expected_msg):
+        validator.validate_from(str(testfile))
+
+
+def test_validator_entrypoint_simple(tmp_path):
+    """'entrypoint' param: simple validation."""
+    tmp_path = pathlib.Path(str(tmp_path))  # comparisons below don't work well in Py3.5
+    testfile = tmp_path / 'testfile'
+    testfile.touch(mode=0o777)
+
+    validator = Validator()
+    resp = validator.validate_entrypoint(str(testfile))
+    assert resp == testfile
+
+
+def test_validator_entrypoint_default(tmp_path):
+    """'entrypoint' param: default value."""
+    tmp_path = pathlib.Path(str(tmp_path))  # comparisons below don't work well in Py3.5
+    default_entrypoint = tmp_path / 'src' / 'charm.py'
+    default_entrypoint.parent.mkdir()
+    default_entrypoint.touch(mode=0o777)
+
+    validator = Validator()
+    validator.basedir = tmp_path
+    resp = validator.validate_entrypoint(None)
+    assert resp == default_entrypoint
+
+
+def test_validator_entrypoint_absolutized(tmp_path, monkeypatch):
+    """'entrypoint' param: check it's made absolute."""
+    # change dir to the temp one, where we will have the 'dirX/file.py' stuff
+    dirx = tmp_path / 'dirX'
+    dirx.mkdir()
+    testfile = dirx / 'file.py'
+    testfile.touch(mode=0o777)
+    monkeypatch.chdir(tmp_path)
+
+    validator = Validator()
+    resp = validator.validate_entrypoint('dirX/file.py')
+    testfile = pathlib.Path(str(testfile))  # comparison below don't work well in Py3.5
+    assert resp == testfile
+
+
+def test_validator_entrypoint_expanded():
+    """'entrypoint' param: expands the user-home prefix."""
+    home = pathlib.Path("~").expanduser()
+    testfd, testpath = tempfile.mkstemp(dir=str(home))
```

Comment:
I'd *really* prefer if test suites don't create files outside of the code directory or $TEMP. creating something in my home dir has a high likelyhood of leaving garbage due to failed test, etc.
I'd actually rather not have the test than have one that touches $HOME.
If there is a different way to test the functionality (set $HOME to somewhere in $TEMP), I'd be ok with it.

Reply:
Ah, awesome idea of touching $HOME!!! 

Fixed the code to use that, so no more files outside temp dir. 

Thanks!!